modules/drivers/mongo/src/metabase/driver/mongo/query_processor.clj

```diff
@@ -549,6 +549,44 @@
                     :unit      unit
                     :amount    amount}})
 
+(defmethod ->rvalue :datetime-diff [[_ x y unit]]
+  (do
+    (check-date-operations-supported)
+    (let [x (->rvalue x)
+          y (->rvalue y)]
+      (case unit
+        :year
+        {$divide [(->rvalue [:datetime-diff x y :month]) 12]}
+
+        :quarter
+        {$divide [(->rvalue [:datetime-diff x y :month]) 3]}
+
+        :week
+        {$divide [(->rvalue [:datetime-diff x y :day]) 7]}
```

Comment:
We do a little extra work calling `->rvalue` twice on `x` and `y`, I wonder if it makes sense to separate the recursive cases from the terminal ones.

Reply:
True. Updated https://github.com/metabase/metabase/pull/27042/commits/51f1529d991279a0c3d437d13a2c11933de0396d